app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -29,6 +31,37 @@ def send_code
 
     private
 
+    def lock_out_max_2fa_sends
+      return unless exceeded_otp_send_limit?
+
+      UpdateUser.new(
+        user: current_user,
+        attributes: {
+          second_factor_locked_at: Time.zone.now,
+          otp_first_sent_at: nil,
+          otp_send_count: 0,
+        }
+      ).call
+
+      handle_second_factor_locked_user(:otp)
```

Comment:
Yeah you're right, it woudl be confusing with the same lockout event, I'll make another.

I don't recall the discussion about not locking the user out, but happy to change that behavior if we want